openfl/component/director/director.py

```diff
@@ -6,50 +6,282 @@
+            self.__dict = {}
+        else:
+            self.__dict = {
+                exp.name: exp
+                for exp in experiments
+            }
+            self.__experiments_queue = list(self.__dict.keys())
+
+    @property
+    def active_experiment(self) -> Union[Experiment, None]:
+        """Get active experiment."""
+        if self.__active_experiment is None:
+            return None
+        return self.__dict[self.__active_experiment]
+
+    @property
+    def queue(self) -> List[str]:
+        """Get queue of not started experiments."""
+        return self.__experiments_queue
+
+    def add(self, experiment: Experiment) -> None:
+        """Add experiment to queue of not started experiments."""
+        self.__dict[experiment.name] = experiment
+        self.__experiments_queue.append(experiment.name)
+
+    def remove(self, name: str) -> None:
+        """Remove experiment from everywhere."""
+        if self.__active_experiment == name:
+            self.__active_experiment = None
+        if name in self.__experiments_queue:
+            self.__experiments_queue.remove(name)
+        if name in self.__archived_experiments:
+            self.__archived_experiments.remove(name)
+        if name in self.__dict:
+            del self.__dict[name]
+
+    async def get_envoy_experiment(self, envoy_name: str) -> str:
+        """Get experiment name for envoy."""
+        queue = self.__col_exp_queues[envoy_name]
+        return await queue.get()
+
+    async def set_next_experiment(self) -> None:
+        """Set next experiment from the queue."""
+        while True:
+            if self.active_experiment is not None or not self.queue:
+                await asyncio.sleep(10)
+                continue
+            self.__active_experiment = self.__experiments_queue.pop(0)
+            return
+
+    async def start_active(self) -> None:
+        """Start active experiment."""
+        loop = asyncio.get_event_loop()
+        run_aggregator = loop.create_task(self.active_experiment.start(
+            tls=self.tls,
+            root_certificate=self.root_certificate,
+            certificate=self.certificate,
+            private_key=self.private_key,
+        ))
+        for col_name in self.active_experiment.collaborators:
+            queue = self.__col_exp_queues[col_name]
+            await queue.put(self.active_experiment.name)
+        await run_aggregator
+        self._finish_active()
+
+    def __getitem__(self, key: str) -> Experiment:
+        """Get experiment by name."""
+        return self.__dict[key]
+
+    def get(self, key: str, default=None) -> Experiment:
+        """Get experiment by name."""
+        return self.__dict.get(key, default)
+
+    def __contains__(self, key: str) -> bool:
+        """Check if experiment exists."""
+        return key in self.__dict
+
+    def _finish_active(self) -> None:
+        self.__dict[self.__active_experiment].__aggregator_grpc_server = None
```

Comment:
Should we explicitly do `del self.__dict[self.__active_experiment].__aggregator_grpc_server` here?

Reply:
If you mean `del __aggregator_grpc_server` it sets to `None`.
If you mean `del self.__dict[self.__active_experiment]` Here we store all information about experiment in `dict[experiment_name] = Experiment()`. If we want to have access to this experiment when it was finished we shouldn't delete this data.